internal/factsengine/gatherers/packageversion.go

```diff
@@ -3,38 +3,50 @@ package gatherers
 import (
 	log "github.com/sirupsen/logrus"
 	"github.com/trento-project/agent/internal/factsengine/entities"
+	"github.com/trento-project/agent/internal/utils"
 )
 
 const (
 	PackageVersionGathererName = "package_version"
 )
 
+// nolint:gochecknoglobals
+var (
+	PackageVersionCMDError = entities.FactGatheringError{
+		Type:    "package-version-cmd-error",
+		Message: "error getting version of package",
+	}
+)
+
 type PackageVersionGatherer struct {
-	executor CommandExecutor
+	executor utils.CommandExecutor
 }
 
 func NewDefaultPackageVersionGatherer() *PackageVersionGatherer {
-	return NewPackageVersionGatherer(Executor{})
+	return NewPackageVersionGatherer(utils.Executor{})
 }
 
-func NewPackageVersionGatherer(executor CommandExecutor) *PackageVersionGatherer {
+func NewPackageVersionGatherer(executor utils.CommandExecutor) *PackageVersionGatherer {
 	return &PackageVersionGatherer{
 		executor: executor,
 	}
 }
 
-func (g *PackageVersionGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.FactsGatheredItem, error) {
-	facts := []entities.FactsGatheredItem{}
+func (g *PackageVersionGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
 	log.Infof("Starting Package versions facts gathering process")
 
 	for _, factReq := range factsRequests {
+		var fact entities.Fact
 		version, err := g.executor.Exec(
 			"rpm", "-q", "--qf", "%{VERSION}", factReq.Argument)
 		if err != nil {
-			// TODO: Decide together with Wanda how to deal with errors. `err` field in the fact result?
-			log.Errorf("Error getting version of package: %s", factReq.Argument)
+			gatheringError := PackageVersionCMDError.Wrap(factReq.Argument)
+			fact = entities.NewFactGatheredWithError(factReq, gatheringError)
+		} else {
+			fact = entities.NewFactGatheredWithRequest(factReq, entities.ParseStringToFactValue(string(version)))
```

Comment:
I don't think we need the `ParseString..` here. We already know that this is a string.
I would put `&entities.FactValueString{Value: string(version)}` here